hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el

```diff
@@ -38,3 +38,87 @@ ANNOTATION: "Waived the assignment as the input is wired to a parameter"
+// 1) reqfifo_rvalid --> 1
+// 2) rspfifo_rvalid --> 1
+// 3) reqfifo_rdata.op --> opread
+// 4) reqfifo_rdata.error --> 1
+//
+// We cannot see vld_rd_rsp && d_error. If d_error is true, we must have rspfifo_rdata.error or
+// reqfifo_rdata.error. The first cannot happen, because the items in the response fifo have an
+// error signal from rerror_i[1] and rom_ctrl connects rerror_i to zero. For the second to happen,
+// we must have pushed in an item with error = 1, which means that error_internal was set. But that
+// will have squashed the request, so req_o was low. As a result, the request fifo item will be
+// popped again before anything appears in the response fifo.
+Condition 24 "1059982851" "(vld_rd_rsp & ((~d_error))) 1 -1" (2 "10")
+
+// The reason for condition 24 explains why we can't get (vld_rd_rsp && reqfifo_rdata.error) ---> 1
+// as if vld_rd_rsp = 1, this means that rspfifo_rvalid = 1. But when we have reqfifo_rdata.error,
+// the request from the req_fifo gets popped again before becoming visible to response fifo. This
+// also explains why we can't expect 1 and 11 for condition 25 and 26 respectively.
+Condition 25 "2807788926" "((vld_rd_rsp && reqfifo_rdata.error) ? error_blanking_integ :
+                            (vld_rd_rsp ? rspfifo_rdata.data_intg :
+                            prim_secded_pkg::SecdedInv3932ZeroEcc)) 1 -1" (2 "1")
+Condition 26 "561780173" "(vld_rd_rsp && reqfifo_rdata.error) 1 -1" (3 "11")
+
+// To cover case 01 for condition 34, both d_valid and d_error requires reqfifo_rvalid. If
+// reqfifo_rvalid is false, both d_valid and d_error must be false. To set d_valid, we need:
+// ((reqfifo_rvalid && ((reqfifo_rdata.op == OpRead && rspfifo_rvalid) || reqfifo_rdata.error)) ||
+//  reqfifo_rvalid).
+// | reqfifo_rvalid | reqfifo_rdata.op | reqfifo_rdata.error | rspfifo_rvalid | d_valid
+// |       1        |        X         |          1          |       X        |    1
+// |       1        |      OpRead      |          0          |       1        |    1
+// |       1        |     !OpRead      |          0          |       X        |    1
+//
+// To set d_error, we need:
+// (reqfifo_rvalid && ((reqfifo_rdata.op == OpRead && (rspfifo_rdata.error | reqfifo_rdata.error))
+//  || reqfifo_rdata.error)).
+// | reqfifo_rvalid | reqfifo_rdata.op | reqfifo_rdata.error | d_error
+// |       1        |      OpRead      |          1          |    1
+// |       1        |     !OpRead      |          1          |    1
+//
+// Excluded rspfifo_rdata.error as it is wired to 0. For rspfifo_rdata.error, it is pushed in
+// rspfifo_wdata.error. If the error field is true, it means rerror_i[1] bit is true but the
+// rerror_i bits are wired to 0 in the instantiation of the tlul_adapter_sram in rom_ctrl.sv.
+//
+// For d_error to set, the min requirement is to get reqfifo_rvalid = 1 and reqfifo_rdata.error = 1
+// but that also set d_valid.
+Condition 34 "201396280" "(d_valid && d_error) 1 -1" (1 "01")
```

Comment:
This took me some time to get my head around properly! But I think the following text is essentially the same proof and might be a bit easier to follow?
```
// We cannot see d_valid = 0 and d_error = 1. For d_error to be true, we need reqfifo_rvalid to be
// true. But then the only way for d_valid to be false is if rspfifo_rvalid is false (and
// reqfifo_rdata.op is OpRead).
//
// The ROM always services TL operations without back-pressure and responds in one cycle. As a
// result, after a request is be pushed into u_reqfifo, it is always popped again on the next cycle.
// As a result, if reqfifo_rvalid is true then rspfifo_wvalid will also be true and (because
// u_rspfifo is in pass-through mode) rspfifo_rvalid will be true.
```